llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -247,6 +247,60 @@ def IREELinalgExt_FftOp : IREELinalgExt_Op<"fft", [
   }];
 }
 
+def IREELinalgExt_ScanOp : IREELinalgExt_Op<"scan", [
+  DeclareOpInterfaceMethods<
+    TiledOpInterface,
+    ["generateScalarImplementation", "getTiledImplementation"]>]> {
+  let summary = "Scan operator";
+  let description = [{
+    Computes the inclusive/exclusive scan along a given dimension.
+  }];
+
+  let arguments = (ins Variadic<AnyType>:$inputs,
+                       Variadic<AnyShaped>:$outputs,
+                       OptionalAttr<I64Attr>:$dimension,
+                       OptionalAttr<I1Attr>:$inclusive
+  );
+  let results = (outs Variadic<AnyRankedTensor>:$results);
+  let regions = (region AnyRegion:$region);
+  let assemblyFormat = [{
+    (`dimension` `(` $dimension^ `)`)?
+    (`inclusive` `(` $inclusive^ `)`)?
+    attr-dict (`ins` `(` $inputs^ `:` type($inputs) `)`)?
+    `outs` `(` $outputs `:` type($outputs) `)`
+    $region (`->` type($results)^)?
+  }];
+
+  let extraClassDeclaration = extraLinalgExtOpClassDeclaration # [{
+    Value input() {
+      return getInputOperand(0)->get();
+    }
+    Value output() {
+      return getOutputOperand(0)->get();
+    }
+    ShapedType getOperandType() {
+      return input().getType().cast<ShapedType>();
+    }
+    int64_t getOperandRank() {
+      return getOperandType().getRank();
+    }
+    uint64_t getScanDimension() {
```

Comment:
If you make this required always, then you can drop these methods `getScanDimension()` can just be `dimension()` and `isInclusive()` will just be `inclusive()`